doc/Variables.md

```diff
@@ -0,0 +1,117 @@
+
+A concrete example will help explain these concepts. We have two ground robots, cleverly named R1 and R2, and we want
+to estimate the pose of both robots as they drive around. Thus, the physical property of interest is the 2D pose. A
+2D pose consists of three floating-point quantities: `x`, `y`, and `'yaw`. Thus our Variable type must include a
+3-tuple to hold the variable's _value_. Now, what about the _identity_ portion of our Variable type? Well, we have two
+different robots, so we need to differentiate between the pose of R1 and the pose of R2. Thus our Variable type must
+include some sort of `robot_id` field. Different instances of our variable will set the `robot_id` field to either 'R1'
+or 'R2'. Additionally, the robots move around over time. So the pose of a robot is not valid for _any_ time, but
+represents the pose of the robot at a _specific_ time. And so our Variable type must also include a `timestamp` field
+to differentiate a pose at time `t1` for a pose at time `t2`.
+
+Just to reiterate the important concepts:
+
+* A variable type defines a _value_ that may be one or more floating-point quantities.
+* A variable instance's _value_ is unknown and/or uncertain. It will change over time as evidence is added.
+* A variable type also defines additional properties that make up its _identity_.
+* A variable instance's _identity_ is immutable.
+* The _identity_ is independent of the _value_.
+
+## Variable Design Principles
+
+The most important consideration when creating a new Variable type is deciding what data dimensions should be
+included. Including too many physical properties in a Variable type will be inefficient and cumbersome when used in
+places where most of the Variable's value dimensions are unused. However, including too few physical properties in
+a Variable type also leads to inefficient and cumbersome usage when even the simplest of observation models involve
+many variables. This is one of those "Goldilocks principle" situations.
+
+![Goldilocks principle](http://home.netcom.com/~swansont_2/goldilocks.jpg)
+
+Understanding how Variable types interact with the rest of the system will help in the design of "good" Variable
+types.
+
+1. The fuse stack is designed to combine observations _of the same variable instance_ from multiple sources. In
+   the robotics example above, one could imagine that the robots measure their current pose using wheel odometry as
+   well as with a laserscan -- two different measurements of the same pose. In order for these two measurements to
+   be combined together in fuse, **they must use the same variable type**. This means that if a variable type defines
+   a 15-dimensional, but a sensor only measures 3 of those dimensions, the sensor must still consume all 15 dimensions
+   and define the measurement error for all 15 dimensions (which will likely mean specifying 12 dimensions worth
```

Comment:
So I can see this point causing confusion, at least until they get to point (2). But point (2) makes things very clear.